Source/UnitTests/Common/FixedSizeQueueTest.cpp

```diff
@@ -38,18 +43,48 @@ TEST(FixedSizeQueue, RingBuffer)
   FixedSizeQueue<int, 5> q;
 
   EXPECT_EQ(0u, q.size());
+  EXPECT_EQ(5u, q.max_size());
 
   q.push(0);
+  EXPECT_EQ(0, q[0]);
   q.push(1);
+  EXPECT_EQ(1, q[1]);
   q.push(2);
   q.push(3);
   q.push(4);
-  q.push(5);
+  EXPECT_EQ(4, q[4]);
+  q.push(5); // Loops to index 0
+  EXPECT_EQ(5, q.beginning());
+  EXPECT_EQ(1, q[0]);
 
   EXPECT_EQ(5u, q.size());
-  EXPECT_EQ(1, q.pop_front());
+  EXPECT_EQ(5u, q.max_size());
 
+  EXPECT_EQ(1, q.pop_front());
   EXPECT_EQ(4u, q.size());
+  q.erase(2);
+  EXPECT_EQ(2u, q.size());
+  q.erase(2);
+  EXPECT_EQ(0u, q.size());
+
+  // Copy over an array bigger than us, which means the copy will loop over
```

Comment:
This should be in another test ("Copy"?).

Reply:
ok